pkg/kubectl/resource/builder.go

```diff
@@ -584,6 +584,9 @@ func (b *Builder) visitorResult() *Result {
 			if b.flatten {
 				visitors = NewFlattenListVisitor(visitors, b.mapper)
 			}
+			if b.defaultNamespace {
+				visitors = NewDecoratedVisitor(visitors, SetNamespace(b.namespace))
+			}
```

Comment:
Because in the label command, it invokes RetrieveLatest function, which needs object's namespace, we should invoke SetNamespace before RetrieveLatest, otherwise it will return an error message ""no namespace set on resource %s %q""